### PR TITLE
Atom mapping tests

### PR DESCRIPTION
**Pull request by richardjgowers**

### CODE REVIEW

File: openfe/setup/atom_mapping/ligandatommapping.py [hunk 1]
```diff
@@ -13,29 +13,29 @@
...
         molB: SmallMoleculeComponent,
-        molA_to_molB: Dict[int, int],
-        annotations: Optional[Dict[str, Any]] = None,
+        molA_to_molB: dict[int, int],
+        annotations: Optional[dict[str, Any]] = None,
     ):
+        """
+        Parameters
+        ----------
+        molA, molB : SmallMoleculeComponent
```
[the remark this answers is not in the document]
Comment (richardjgowers) [reply openfe/setup/atom_mapping/ligandatommapping.py:27]: Yeah actually I was wrestling with this in this PR.  I had refactored everything to use componentX... should I put that back in?

[the remark this answers is not in the document]
Comment (IAlibay) [reply openfe/setup/atom_mapping/ligandatommapping.py:27]: I'd say yes - if the methods say component then we have to go for a full buy-in imho

[the remark this answers is not in the document]
Comment (richardjgowers) [reply openfe/setup/atom_mapping/ligandatommapping.py]: yeah ok you're right, let's just get this over with while we can

**Review by IAlibay — APPROVED** [review]
lgtm

**Pull request merged**